Sources/BrowserServicesKit/ContentBlocking/ClickToLoadRulesSplitter.swift

```diff
@@ -0,0 +1,151 @@
+//
+//  ClickToLoadRulesSplitter.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import TrackerRadarKit
+
+public struct ClickToLoadRulesSplitter {
+
+    public enum Constants {
+
+        public static let clickToLoadRuleListPrefix = "CTL_"
+        public static let tdsRuleListPrefix = "TDS_"
+
+    }
+
+    private let rulesList: ContentBlockerRulesList
+
+    public init(rulesList: ContentBlockerRulesList) {
+        self.rulesList = rulesList
+    }
+
+    public func split() -> (withoutBlockCTL: ContentBlockerRulesList, withBlockCTL: ContentBlockerRulesList)? {
+        var withoutBlockCTL: (tds: TrackerData, etag: String)?
+        var withBlockCTL: (tds: TrackerData, etag: String)?
+
+        if let trackerData = rulesList.trackerData {
+            guard let splitTDS = split(trackerData: trackerData) else { return nil }
+
+            withoutBlockCTL = splitTDS.withoutBlockCTL
+            withBlockCTL = splitTDS.withBlockCTL
+        }
+
+        guard let splitFallbackTDS = split(trackerData: rulesList.fallbackTrackerData) else { return nil }
```

Comment:
This is not how we want it - current code processes both TDS (current and fallback) while in 99.9% of the cases only one is needed. So we should be processing fallback only when needed - that's the reason we use autoclosures in `ContentBlockerRulesList`